agreement/player.go

```diff
@@ -111,7 +125,8 @@ func (p *player) handle(r routerHandle, e event) []action {
 			delta := time.Duration(e.RandomEntropy % uint64(upper-lower))
 
 			p.Napping = true
-			p.Deadline = lower + delta
+			p.Deadline.Duration = lower + delta
+			p.Deadline.Type = TimeoutDeadline
```

Comment:
all of these and below need to be initialized as a whole
```suggestion
			p.Deadline = Deadline{Duration: lower + delta, Type: TimeoutDeadline}
```